src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java

```diff
@@ -208,4 +223,61 @@ public static Types.MapType mapFromJson(JsonNode json) {
       return Types.MapType.ofOptional(keyId, valueId, keyType, valueType);
     }
   }
+
+  private static org.apache.parquet.schema.Type replaceWithOriginalFieldName(
+      org.apache.parquet.schema.Type parquetType, Type icebergType, String fieldName) {
+    if (parquetType.isPrimitive() != icebergType.isPrimitiveType()
+        || (!parquetType.isPrimitive()
+            && parquetType.getLogicalTypeAnnotation()
+                == null /* ignore outer layer of map or list */
+            && parquetType.asGroupType().getFieldCount()
+                != icebergType.asNestedType().fields().size())) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Parquet type and Iceberg type mismatch: %s, %s", parquetType, icebergType));
```

Comment:
"Parquet type and Iceberg type mismatch. parquetType=%s, icebergType=%s"

Reply:
Updated